packages/engine/paima-funnel/src/reading.ts

```diff
@@ -178,21 +180,29 @@ export async function fetchDynamicEvmPrimitives(
     for (const _ext of exts) {
       const ext: CdeDynamicEvmPrimitiveDatum = _ext as CdeDynamicEvmPrimitiveDatum;
 
+      const cdeName = `${ext.cdeName}-${
```

Comment:
We should use re-use a utility function we expose publicly to combine a base name with an ID

i.e.

```js
generateDynamicPrimitiveName(
    ext.cdeName,
    sharedData.extensions.filter(
          e =>
            e.cdeType !== ChainDataExtensionType.DynamicEvmPrimitive &&
            e.cdeName.startsWith(ext.cdeName)
        ).length
)
```